### PR TITLE
K8SPSMDB-548: Add PSMDB 5.0 support

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPSMDB-548](https://jira.percona.com/browse/K8SPSMDB-548) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/1e53e201438ef87e7c9addd7be46ea73714dca22
image: `perconalab/percona-server-mongodb-operator:PR-757-1e53e201`

**Review by delgod — APPROVED** [review]

**Review by delgod — APPROVED** [review]

File: healthcheck/tools/db/db.go [hunk 1]
```diff
@@ -15,110 +15,65 @@
...
-			"source": cnf.DialInfo.Source,
-		}).Debug("Enabling authentication for session")
-	}
+	opts := options.Client().
+		SetHosts(conf.Hosts).
+		SetReplicaSet(conf.ReplSetName).
+		SetAuth(options.Credential{Password: conf.Password, Username: conf.Username}).
+		SetTLSConfig(conf.TLSConf).
+		SetServerSelectionTimeout(2 * time.Second).
+		SetConnectTimeout(3 * time.Second)
```
Comment (egegunes) [inline healthcheck/tools/db/db.go:45]: @hors what do you think about these timeouts?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/50aa2f8c3798138077cfe7545f092f211e6ce17e
image: `perconalab/percona-server-mongodb-operator:PR-757-50aa2f8c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/0054b40c5e78561a3a265f84cf33a3c87130fd3b
image: `perconalab/percona-server-mongodb-operator:PR-757-0054b40c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/ce6d0e45dbaad6b196324350f3600610d7e4995d
image: `perconalab/percona-server-mongodb-operator:PR-757-ce6d0e45`

File: cmd/mongodb-healthcheck/main.go [hunk 2]
```diff
@@ -128,14 +141,18 @@ func main() {
...
 			os.Exit(1)
 
 		case "mongos":
-			err := healthcheck.MongosReadinessCheck(session)
+			err := healthcheck.MongosReadinessCheck(client)
 			if err != nil {
 				log.Error(err.Error())
-				session.Close()
+				if err := client.Disconnect(context.TODO()); err != nil {
+					log.Errorf("failed to disconnect: %v", err)
```
Comment (nonemax) [inline cmd/mongodb-healthcheck/main.go:154]: maybe it's better to put  something like 'errorMessage' into `if` and after it call log.Errorf(...

File: healthcheck/tools/db/ssl.go [hunk 3]
```diff
@@ -52,103 +48,53 @@ func LastSSLError() error {
...
-			err = validateConnection(conn, config, dnsName)
-			if err != nil {
-				log.Errorf("Could not disable hostname validation. Got: %v", err)
-			}
-		}
-
-		return conn, err
+	pemOk, err := isFileExists(cnf.SSL.PEMKeyFile)
+	if err != nil {
+		return fmt.Errorf("Failed to check if file with name %s exists, err: %v", cnf.SSL.PEMKeyFile, err)
```
Comment (nonemax) [inline healthcheck/tools/db/ssl.go:62]: I think it's better to use `errors` pkg here and below

**Review by nonemax — CHANGES_REQUESTED** [review]

File: healthcheck/tools/db/ssl.go (hunk 3 above)
Comment (egegunes) [reply healthcheck/tools/db/ssl.go]: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/b7032b2da10a4b4eb30909772ea64bb56beb9f33

File: cmd/mongodb-healthcheck/main.go (hunk 2 above)
Comment (egegunes) [reply cmd/mongodb-healthcheck/main.go]: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/b7032b2da10a4b4eb30909772ea64bb56beb9f33

**Review by delgod — APPROVED** [review]

File: healthcheck/tools/db/db.go (hunk 1 above)
Comment (hors) [reply healthcheck/tools/db/db.go:45]: @egegunes  it is not so simple to choose right timeouts. I think we need to ask experts about it. But for now we can avoid setting them and as soon as we have right numbers (consulting with experts can take some time) we will set them.

**Review by hors — CHANGES_REQUESTED** [review]

File: healthcheck/tools/db/db.go (hunk 1 above)
Comment (egegunes) [reply healthcheck/tools/db/db.go]: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/51d5562b7016f66c33a6b43e707c68ecbe6d2390

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| version-service | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/cd2ef823010e6f0c616c5d3260fa41c48cecdc2a
image: `perconalab/percona-server-mongodb-operator:PR-757-cd2ef823`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scheduled-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/51d5562b7016f66c33a6b43e707c68ecbe6d2390
image: `perconalab/percona-server-mongodb-operator:PR-757-51d5562b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| pitr | passed |
| pitr-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/2bd15a8b362ff8f6d8effc36ee5f51ec90c4a82d
image: `perconalab/percona-server-mongodb-operator:PR-757-2bd15a8b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/757/commits/c37d53904195028fff29796834324679873f0120
image: `perconalab/percona-server-mongodb-operator:PR-757-c37d5390`

**Review by nonemax — APPROVED** [review]

**Pull request merged**